application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -0,0 +1,144 @@
+                .setEphemeral(true)
+                .queue();
+
+            logger.error("The bot does not have BAN_MEMBERS permission on the server '{}' ",
+                    guild.getName());
+            return false;
+        }
+        return true;
+    }
+
+    @SuppressWarnings("MethodWithTooManyParameters")
+    private static void banUser(@NotNull User target, @NotNull Member author,
+            @NotNull String reason, int deleteHistoryDays, @NotNull Guild guild,
+            @NotNull SlashCommandEvent event) {
+        event.getJDA()
+            .openPrivateChannelById(target.getIdLong())
+            .flatMap(channel -> channel.sendMessage(
+                    """
+                            Hey there, sorry to tell you but unfortunately you have been banned from the server %s.
+                            If you think this was a mistake, please contact a moderator or admin of the server.
+                            The reason for the ban is: %s
+                            """
+                        .formatted(guild.getName(), reason)))
+            .mapToResult()
+            .flatMap(result -> guild.ban(target, deleteHistoryDays, reason))
+            .flatMap(result -> event.reply("'%s' was banned by '%s' for: %s"
+                .formatted(target.getAsTag(), author.getUser().getAsTag(), reason)))
+            .queue();
+
+        logger.info(
+                "'{}' ({}) banned the user '{}' ({}) from guild '{}' and deleted their message history of the last {} days, for reason '{}'",
+                author.getUser().getAsTag(), author.getId(), target.getAsTag(), target.getId(),
+                guild.getName(), deleteHistoryDays, reason);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(TARGET_OPTION), "The target is null");
+        Member target = targetOption.getAsMember();
+        Member author = Objects.requireNonNull(event.getMember(), "The author is null");
+        String reason = Objects.requireNonNull(event.getOption(REASON_OPTION), "The reason is null")
+            .getAsString();
+
+        Guild guild = Objects.requireNonNull(event.getGuild());
+        Member bot = guild.getSelfMember();
+
+        // Member doesn't exist if attempting to ban a user who is not part of the guild.
+        if (target != null && !handleCanInteractWithTarget(bot, author, target, event)) {
+            return;
+        }
+        if (!handleHasPermissions(bot, author, guild, event)) {
+            return;
+        }
+        if (!ModerationUtils.handleReason(reason, event)) {
+            return;
+        }
+
+        int deleteHistoryDays = Math
+            .toIntExact(Objects.requireNonNull(event.getOption(DELETE_HISTORY_OPTION)).getAsLong());
+        banUser(targetOption.getAsUser(), author, reason, deleteHistoryDays, guild, event);
```

Comment:
Might be a good idea to add a gap for readability
```suggestion
         int deleteHistoryDays = Math
            .toIntExact(Objects.requireNonNull(event.getOption(DELETE_HISTORY_OPTION)).getAsLong());
            
        banUser(targetOption.getAsUser(), author, reason, deleteHistoryDays, guild, event);
```

Reply:
Fixed